### PR TITLE
Remove setting net.Conn Deadlines as its not needed anymore

### LINKED ISSUE #7852: PHP S3 Stream Wrapper Hangs for 2 minutes in latest Minio
I am using the latest Docker minio image. Last week, I was testing using PHP S3 Stream Wrapper and everything worked fine. I don't know the Minio version I was using last week, as I hadn't pulled new image of minio from the docker hub for a few weeks. Tonight, I pulled a new minio image and resumed testing especially testing PHP S3 Stream Wrapper (using the AWS PHP SDK), and tonight, Minio takes 2 minutes to complete the file_get_contents( "s3://we-bucket/t.txt" ) for a very small text file. Using getObject() works fine.

Also, running the same tests using an identical AWS S3 bucket, works fine for both s3:// wrapper and for getObject() calls.  Minio appears to be hanging for 1 minute before printing out the file contents and then continues to hang for 1 minute more (for a total of 2 minutes) before returning from the file_get_contents() function. Same for fread() and readfile().

So, maybe some sort of regression with s3 stream wrapper?

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Remove setting net.Conn Deadlines as it's not needed anymore

## Motivation and Context
This commits fixes a bug introduced in af6c6a2.

Setting deadlines in Go results in arbitrary hangs as reported here
golang/go#34385

Fixes https://github.com/minio/minio/issues/7852

## How to test this PR?
Just the code shared in the reproducer 

```php
<?php

// Include the SDK using the Composer autoloader
date_default_timezone_set('America/Los_Angeles');
require 'vendor/autoload.php';

$s3 = new Aws\S3\S3Client([
        'version' => 'latest',
        'region'  => 'us-east-1',
        'endpoint' => 'http://127.0.0.1:9001',
        'use_path_style_endpoint' => true,
        'credentials' => [
                'key'    => 'minio',
                'secret' => 'minio123',
        ],
]);

$path = "s3://testbucket/hosts";
        
$s3->registerStreamWrapper();

$data = file_get_contents($path);

// Print the body of the result by indexing into the result object.
print_r( $data );
```
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression introduced in af6c6a2b35ecd67300739f7ecb54fad60eeb4967
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by balamurugana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by nitisht** [comment]
waiting for mint-auto test run before merge

**Comment by kannappanr** [comment]
@balamurugana @vadmeste Can you please review again?

**Review by balamurugana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#771c1b70-786f-4b22-9df5-f66aa9b78019) |
<a name="771c1b70-786f-4b22-9df5-f66aa9b78019"></a>
### 8269-a94e0a3/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31327
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8eb42a0d0506:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 13 seconds
(5/14) Running awscli tests ... done in 2 minutes and 23 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 1 minutes and 13 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 42 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 36 seconds
{
  "args": {},
  "duration": 99640,
  "error": "The specified key does not exist.",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "Stat failed",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :x: [more...](#0b59de13-5c29-4796-b5b5-40de101c93f6) |
| mint-fs.sh | :x: [more...](#12ccc161-8ffd-4e52-bb3c-82e86a1098dc) |
<a name="0b59de13-5c29-4796-b5b5-40de101c93f6"></a>
### 8269-ff2be85/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32089
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 745035fe6c7c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 2 minutes and 11 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 0 seconds
(8/14) Running minio-dotnet tests ... done in 56 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 15 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 51 seconds
(11/14) Running minio-js tests ... FAILED in 30 minutes and 14 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 30,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:83:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:153:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:77:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:607:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:616:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:629:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:567:22) at IncomingMessage.onend (_stream_readable.js:514:10) at endReadableNT (_stream_readable.js:978:12) at _combinedTickCallback (internal/process/next_tick.js:80:11) at process._tickCallback (internal/process/next_tick.js:104:9)"
}

Executed 10 out of 14 tests successfully.
```
<a name="12ccc161-8ffd-4e52-bb3c-82e86a1098dc"></a>
### 8269-ff2be85/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:32155
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e312ca1e45c2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 56 seconds
(8/14) Running minio-dotnet tests ... done in 29 seconds
(9/14) Running minio-go tests ... done in 47 seconds
(10/14) Running minio-java tests ... done in 54 seconds
(11/14) Running minio-js tests ... done in 47 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 32 seconds
{
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "duration": 104,
  "status": "FAIL",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "message": "AccessDenied: message: Access Denied",
  "args": {
    "response.reason": "Forbidden",
    "bucket_name": "minio-py-test-a05b5b60-37e0-492d-b84b-6b019501667c",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-20T03-26-09Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C608E25C2CB304\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Fri, 20 Sep 2019 03:48:08 GMT\"]}}",
    "response.status": 403,
    "object_name": "43547ea6-6ebb-41f6-ac1e-33db0be88a39",
    "presigned_get_object_url": "http://72.28.97.58:32155/minio-py-test-a05b5b60-37e0-492d-b84b-6b019501667c/43547ea6-6ebb-41f6-ac1e-33db0be88a39?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190920%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190920T034803Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=995fac48aebc4a061ff9c3ea3dadef11856c9759bdcb8af60f864d7df162ac21",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>43547ea6-6ebb-41f6-ac1e-33db0be88a39</Key><BucketName>minio-py-test-a05b5b60-37e0-492d-b84b-6b019501667c</BucketName><Resource>/minio-py-test-a05b5b60-37e0-492d-b84b-6b019501667c/43547ea6-6ebb-41f6-ac1e-33db0be88a39</Resource><RequestId>15C608E25C2CB304</RequestId><HostId>c15e53f2-f4f8-4aeb-be8d-7183a2beee9a</HostId></Error>"
  }
}

Executed 11 out of 14 tests successfully.
```

File: cmd/http/accountingconn.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type AccountingConn struct {
+	net.Conn
+	updateBytesReadFunc    func(int) // function to be called to update bytes read.
+	updateBytesWrittenFunc func(int) // function to be called to update bytes written.
+}
+
+// Read - reads data from the connection using wrapped buffered reader.
+func (c *AccountingConn) Read(b []byte) (n int, err error) {
+	n, err = c.Conn.Read(b)
+	if err == nil && c.updateBytesReadFunc != nil {
```
Comment (vadmeste) [inline cmd/http/accountingconn.go:33]: Let's update read bytes even if err is not nil, I don't think Read() always returns n == 0 when err is not nil, (connection closed unexpectedly ?)

File: cmd/http/accountingconn.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		c.updateBytesReadFunc(n)
+	}
+
+	return n, err
+}
+
+// Write - writes data to the connection.
+func (c *AccountingConn) Write(b []byte) (n int, err error) {
+	n, err = c.Conn.Write(b)
+	if err == nil && c.updateBytesWrittenFunc != nil {
```
Comment (vadmeste) [inline cmd/http/accountingconn.go:43]: Let's update write bytes even if err is not nil, I don't think Write() always returns n == 0 when err is not nil, (connection closed unexpectedly ?)

**Review by vadmeste — COMMENTED** [review]
Some comments

File: cmd/http/accountingconn.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/http/accountingconn.go:43]: Short writes are not allowed upon error write returns always 0

File: cmd/http/accountingconn.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/http/accountingconn.go:33]: This code will go away in Prometheus cleanup..

File: cmd/http/accountingconn.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/http/accountingconn.go:43]: This code will go away in prometheus cleanup PR by @Praveenrajmani - I am not trying to fix any orthogonal issues.

**Pull request merged**